### PR TITLE
exported validate and validateValues func

### PR DESCRIPTION
**Pull request by deva26**
We would like to use some parts of istio operator code base in our system. For that a couple of generic validation functions need to be exposed. This is a very local change with minimum impact.

### CODE REVIEW

**Comment by dgn** [comment]
/ok-to-test

I'm fine with the changes but 
a) this might break anytime
b) couldn't you use the already-public func CheckIstioOperatorSpec?

**Comment by deva26** [comment]
> /ok-to-test
> 
> I'm fine with the changes but
> a) this might break anytime
> b) couldn't you use the already-public func CheckIstioOperatorSpec?

Thanks, Daniel. We are trying to use the validation logic for non IstioOperatorSpec as well. We understand this may break anytime.

File: operator/pkg/validate/validate.go [hunk 1]
```diff
@@ -54,10 +54,11 @@ func CheckIstioOperatorSpec(is *v1alpha1.IstioOperatorSpec, checkRequiredFields
 	}
 
 	errs = CheckValues(is.Values)
-	return util.AppendErrs(errs, validate(defaultValidations, is, nil, checkRequiredFields))
+	return util.AppendErrs(errs, Validate(DefaultValidations, is, nil, checkRequiredFields))
 }
 
-func validate(validations map[string]ValidatorFunc, structPtr interface{}, path util.Path, checkRequired bool) (errs util.Errors) {
+// Validate validates the values of the tree using the supplied Func
```
Comment (rshriram) [inline operator/pkg/validate/validate.go]: may be add a note saying this is also used by third party integrations and that it shouldn't be made private?

**Review by richardwxn — APPROVED** [review]

**Pull request merged**